camel/loaders/multi_source_parser.py

```diff
@@ -0,0 +1,212 @@
+
+        This method processes multiple sources in parallel, extracts text
+        content, combines the extracted elements, chunks them based on size,
+        and applies text cleaning operations.
+
+        Args:
+            sources (List[str]): List of file paths or URLs to parse. Each
+                source can be either a local file path or a valid URL.
+            chunk_size (int, optional): Maximum number of characters per chunk.
+                Defaults to 6000.
+            overlap (int, optional): Number of overlapping chunks to create.
+                This helps maintain context between chunks. Defaults to 1.
+
+        Returns:
+            List[str]: A list of cleaned text chunks. Each chunk is processed
+            according to the clean_options specified in the class.
+
+        Note:
+            - If a source fails to parse, a warning message will be printed
+            - If no elements are successfully parsed from any source, an empty
+              list will be returned
+            - Text cleaning is performed in parallel for better performance
+        """
+        all_elements = []
+
+        # Parse all sources in parallel with limited workers
+        with ThreadPoolExecutor(max_workers=self.max_workers) as executor:
+            futures = {
+                executor.submit(self.uio.parse_file_or_url, source): source
+                for source in sources
+            }
+            for future in futures:
+                try:
+                    elements = future.result()
+                    # Only extend if elements were successfully parsed
+                    if elements is not None:
+                        all_elements.extend(elements)
+                    else:
+                        source = futures[future]
+                        logger.warning(
+                            f": No elements extracted from {source}"
+                        )
+                except Exception as e:
+                    source = futures[future]
+                    logger.error(f"Error processing {source}: {e!s}")
+
+        if not all_elements:
+            logger.warning(
+                "No elements were successfully parsed from any\
+            sources"
+            )
+            return []
+
+        # Chunk the elements
+        chunks = self.uio.chunk_elements(
+            elements=all_elements,
+            chunk_type="chunk_by_title",
+            max_characters=chunk_size,
+            overlap=overlap,
+        )
+
+        # Clean each chunk in parallel with limited workers
+        with ThreadPoolExecutor(max_workers=self.max_workers) as executor:
+            cleaned_chunks = list(
+                executor.map(
+                    lambda chunk: self.uio.clean_text_data(
+                        text=str(chunk),
+                        clean_options=self.clean_options,
+                    ),
+                    chunks,
+                )
+            )
```

Comment:
I believe the two steps (parsing and cleaning) can be merged into the same thread pool to reduce the overhead of creating and destroying separate thread pools.

```suggestion
            if not all_elements:
                logger.warning(
                    "No elements were successfully parsed from any\
                sources"
                )
                return []
    
            # Chunk the elements
            chunks = self.uio.chunk_elements(
                elements=all_elements,
                chunk_type="chunk_by_title",
                max_characters=chunk_size,
                overlap=overlap,
            )
    
            # Clean each chunk in parallel with limited workers
            cleaned_chunks = list(
                executor.map(
                    lambda chunk: self.uio.clean_text_data(
                        text=str(chunk),
                        clean_options=self.clean_options,
                    ),
                    chunks,
                )
            )
```